userspace/libsinsp/sinsp_filtercheck_thread.cpp

```diff
@@ -46,29 +46,30 @@ static inline bool str_match_start(const std::string& val, size_t len, const cha
+	{PT_CHARBUF, EPF_ARG_ALLOWED, PF_NA, "proc.acmdline", "Ancestor Command Line", "The full command line (proc.name + proc.args) for a specific process ancestor. You can access different levels of ancestors by using indices. For example, proc.acmdline[1] retrieves the full command line of the parent process, proc.acmdline[2] retrieves the proc.cmdline of the grandparent process, and so on. The current process's full command line can be obtained using proc.acmdline[0]. When used without any arguments, proc.acmdline is applicable only in filters and matches any of the process ancestors. For instance, you can use `proc.acmdline contains base64` to match any process ancestor whose command line contains the term base64."},
 	{PT_UINT64, EPF_NONE, PF_DEC, "proc.cmdnargs", "Number of Command Line args", "The number of command line args (proc.args)."},
-	{PT_UINT64, EPF_NONE, PF_DEC, "proc.cmdlenargs", "Total Count of Characters in Command Line args", "The total count of characters / length of the comamnd line args (proc.args) combined excluding whitespaces between args."},
+	{PT_UINT64, EPF_NONE, PF_DEC, "proc.cmdlenargs", "Total Count of Characters in Command Line args", "The total count of characters / length of the command line args (proc.args) combined excluding whitespaces between args."},
 	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.exeline", "Executable Command Line", "The full command line, with exe as first argument (proc.exe + proc.args) when starting the process generating the event."},
-	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.env", "Environment", "The environment variables of the process generating the event."},
+	{PT_CHARBUF, EPF_ARG_ALLOWED, PF_NA, "proc.env", "Environment", "The environment variables of the process generating the event as concatenated string 'ENV_NAME=value ENV_NAME1=value1'. Can also be used to extract the value of a known env variable, e.g. proc.env[ENV_NAME]."},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "proc.aenv", "Ancestor Environment", "[EXPERIMENTAL] Extract the first matched value of a known env variable among the ancestors (up to 7 levels) of the current process, e.g. proc.aenv[ENV_NAME]. proc.aenv can also be used for filtering the ancestors' proc.env. When used as output field without [ENV_NAME] it returns proc.env. The behavior of this ancestor filtering differs from that in existing fields like proc.aname. For instance, specifying the traversal level is not supported. This field may be deprecated or undergo breaking changes in future releases, such as proc.aenv[1][ENV_NAME]. Please use it with caution."},
```

Comment:
or at this point, we can even change the name to something like `proc.find_aenv[]`

Reply:
I mostly agree with @Andreagit97 

In particular:
- `When used as output field without [ENV_NAME] it returns proc.env`, not clear why we should support this? :thinking: 
- Although `proc.aenv[ENV_NAME]` and `proc.aenv[2]` may work because `2` is not a valid env var identifier, I still found it a bit confusing. Why not using two distinct fields? :thinking: 

I don't want to be a blocker, since it's experimental. However, it would be better if this field was trying to achieve only one use case, instead of multiple

Reply:
>   proc.aenv[ENV_NAME] -> returns the value of the variable
>   proc.aenv[2] -> return the list of the env variables of the second ancestor

yes this can be easily implemented, will do!


> When used as output field without [ENV_NAME] it returns proc.env. This part is not clear to me, why do we want to return proc.env when no argument is provided? maybe I missed something, but I would always force the user to specify an argument here, WDYT?

This seems to be the default behavior if the other fields like `proc.aname` are used without level. This statement should only apply to using the field as output. We could leave it just to stay consistent but I can remove it from the desc?

Reply:
@leogr re 

> Although proc.aenv[ENV_NAME] and proc.aenv[2] may work because 2 is not a valid env var identifier, I still found it a bit confusing. Why not using two distinct fields

Understood, at the same time I wouldn't see problems with it as adopters are very familiar with the numbers indicating parent process level.

Reply:
PTAL at the new description, now flagged as `EPF_ARG_ALLOWED`. Thanks!